test/net_http_test.rb

```diff
@@ -0,0 +1,389 @@
+require 'test_helper'
+require 'semian/net_http'
+require 'thin'
+
+class TestNetHTTP < MiniTest::Unit::TestCase
+  class RackServer
+    def self.call(env)
+      response_code = env['REQUEST_URI'].delete("/")
+      response_code = '200' if response_code == ""
+      [response_code, {'Content-Type' => 'text/html'}, ['Success']]
+    end
+  end
+
+  PORT = 31_050
+  TOXIC_PORT = PORT + 1
+
+  DEFAULT_SEMIAN_OPTIONS = {
+    tickets: 3,
+    success_threshold: 1,
+    error_threshold: 3,
+    error_timeout: 10,
+  }
+
+  def test_semian_identifier
+    with_server do
```

Comment:
Yeah I'm not completely sure what you mean by the hostname part. I did add the ability to pass in multiple ports as an array to `with_server`. https://github.com/Shopify/semian/pull/58#discussion_r43900596
Do you mean that you can technically give it different hostnames to make a server on? How would you handle if they give some hostname like `google.com` which you can't bind a server to? Or can you?